### PR TITLE
SAK-47013 BLTI web portlet can cause an NPE when manually placed in a site

### PR DESCRIPTION
**Pull request by csev**
https://sakaiproject.atlassian.net/browse/SAK-47013

### CODE REVIEW

File: basiclti/web-ifp/src/bundle/vm/pick.vm [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+<p>$tlang.getString("content.no.tools")</p>
+<p>
+<p class="act">
+<input type="submit" accesskey ="x" name="$doCancel" value="$tlang.getString('gen.cancel')" />
+</p>
+#end
+	</form>
+
+<script type="text/javascript">
+$(document).ready(function () {
```
Comment (ern) [inline basiclti/web-ifp/src/bundle/vm/pick.vm:52]: we should stop using jquery's ready
`document.addEventListener("DOMContentLoaded", function() { });`
see https://stackoverflow.com/questions/2304941/what-is-the-non-jquery-equivalent-of-document-ready

File: basiclti/web-ifp/src/bundle/vm/pick.vm [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<script type="text/javascript">
+	focus_path = [ "title" ];
+</script>
+
+<div class="portletBody">
+<script type="text/javascript" src="/library/js/headscripts.js"></script>
```
Comment (ern) [inline basiclti/web-ifp/src/bundle/vm/pick.vm:6]: When possible try to include js at the bottom of the file

File: basiclti/web-ifp/src/java/org/sakaiproject/portlets/SakaiIFrame.java [hunk 3]
```diff
@@ -533,4 +584,6 @@ private Long getLongNull(Object key) {
 		}
 		return null;
 	}
+
+	// vim: set noet ts=4
```
Comment (ern) [inline basiclti/web-ifp/src/java/org/sakaiproject/portlets/SakaiIFrame.java:588]: remove

File: basiclti/web-ifp/src/bundle/vm/edit.vm [hunk 4]
```diff
@@ -33,5 +33,11 @@
 			#end
 		</p>
 	</form>
-<script type="text/javascript">$(document).ready(function () { fontawesome_icon_picker('#fa_icon'); });</script>
+<script type="text/javascript">
+$(document).ready(function () {
```
Comment (ern) [inline basiclti/web-ifp/src/bundle/vm/edit.vm:37]: same comment as below

File: basiclti/web-ifp/pom.xml [hunk 5]
```diff
@@ -31,6 +31,10 @@
             <groupId>org.sakaiproject.kernel</groupId>
             <artifactId>sakai-component-manager</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
```
Comment (jonespm) [inline basiclti/web-ifp/pom.xml:36]: You don't **have** to include commons-lang3 with commons-lang here. commons-lang has the same [NumberUtils.toLong](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/math/NumberUtils.html#toLong(java.lang.String,%20long)) you're using here. 

Though it probably would be worth upgrading to lang3 and removing lang here

File: basiclti/web-ifp/src/bundle/vm/pick.vm (hunk 2 above)
Comment (csev) [reply basiclti/web-ifp/src/bundle/vm/pick.vm:6]: The headscripts.js in Sakai is intended to be included at the beginning.  Hence the name "headscripts" - it is tiny and kept that way for really basic utility code that is often needed in the body of the code.

File: basiclti/web-ifp/src/bundle/vm/pick.vm (hunk 1 above)
Comment (csev) [reply basiclti/web-ifp/src/bundle/vm/pick.vm:52]: OK.

File: basiclti/web-ifp/src/java/org/sakaiproject/portlets/SakaiIFrame.java (hunk 3 above)
Comment (csev) [reply basiclti/web-ifp/src/java/org/sakaiproject/portlets/SakaiIFrame.java:588]: OK

File: basiclti/web-ifp/pom.xml (hunk 5 above)
Comment (csev) [reply basiclti/web-ifp/pom.xml]: Thanks.  I think I will just use lang and leave upgrade to lang3 later so I feel like I don't need to test as much.

File: basiclti/web-ifp/src/bundle/vm/edit.vm (hunk 4 above)
Comment (csev) [reply basiclti/web-ifp/src/bundle/vm/edit.vm]: OK

**Review by ern — APPROVED** [review]

File: basiclti/web-ifp/src/bundle/vm/pick.vm [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    #end
+   </select>
+</p>
+<p class="act">
+<input type="submit" accesskey ="s" class="active" name="$doChoose" value="$tlang.getString('gen.save')" />
+<input type="submit" accesskey ="x" name="$doCancel" value="$tlang.getString('gen.cancel')" />
+</p>
+#else
+<p>$tlang.getString("content.no.tools")</p>
+<p>
```
Comment (fostersdesign) [inline basiclti/web-ifp/src/bundle/vm/pick.vm:44]: Why is this <p> just sitting by itself here unclosed?

```suggestion

```

File: basiclti/web-ifp/src/bundle/vm/pick.vm [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+<p class="act">
+<input type="submit" accesskey ="x" name="$doCancel" value="$tlang.getString('gen.cancel')" />
+</p>
```
Comment (fostersdesign) [inline basiclti/web-ifp/src/bundle/vm/pick.vm:47]: Is this a duplication of line 40? Is it needed?
```suggestion

```

**Review by fostersdesign — COMMENTED** [review]
Some markup changes requested with suggestions.

File: basiclti/web-ifp/src/bundle/vm/pick.vm (hunk 7 above)
Comment (fostersdesign) [reply basiclti/web-ifp/src/bundle/vm/pick.vm:46]: Oh, sorry! I missed the else...

**Pull request merged**